Build of ICE40 bitbang driver fails with PINCTRL

**Describe the bug**
It is impossible to build the ICE40 bitbang driver with PINCTRL=y.

**To Reproduce**

    west build --board nucleo_f746zg zephyr/tests/drivers/build_all/fpga

**Expected behavior**
It should be possible to build the ICE40 bitbang driver with PINCTRL=y.

**Impact**
It is impossible to build the ICE40 bitbang driver with PINCTRL=y.

**Logs and console output**

    .../zephyr/drivers/fpga/fpga_ice40_bitbang.c:232:41: error: 'const struct fpga_ice40_config' has no member named 'pincfg'
      232 |         (void)pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
          |             

**Environment (please complete the following information):**

Zephyr revision 43e8f11d9445afb1aa1328709be6466691374fe3

--- a/drivers/fpga/fpga_ice40_bitbang.c
+++ b/drivers/fpga/fpga_ice40_bitbang.c
@@ -229,7 +229,7 @@ unlock:
 	(void)gpio_pin_configure_dt(&config_bitbang->clk, GPIO_DISCONNECTED);
 	(void)gpio_pin_configure_dt(&config_bitbang->pico, GPIO_DISCONNECTED);
 #ifdef CONFIG_PINCTRL
-	(void)pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	(void)pinctrl_apply_state(config_bitbang->pincfg, PINCTRL_STATE_DEFAULT);
 #endif /* CONFIG_PINCTRL */
 
 	k_spin_unlock(&data->lock, key);

